msc_pygeoapi/process/weather/extract_sounding_data.py

```diff
@@ -0,0 +1,640 @@
+    dew point depression, wind speed, wind direction for each pressure level.
+    Also gives other useful sounding parameters.
+    :param model: model name
+    :param model_run: date and hour at which the model is ran
+    :param forecast_hour: valid hour
+    :param lon: longitude
+    :param lat: latitude
+    :param convection_indices: True or False
+    :param temperature_data: True or False
+    :param wind_data: True or False
+    :param novalues_above_100mbar: If True, stop at 100mbar
+    :return: json containing data for the given params
+    """
+
+    from msc_pygeoapi.process.weather import (
+        GEOMET_DDI_BASEPATH,
+        GEOMET_SCIENCE_BASEPATH,
+    )
+
+    data_basepath = GEOMET_DDI_BASEPATH
+
+    date = datetime.datetime.strptime(model_run, DATE_FORMAT)
+    run_hour = f'{date.hour:02}'
+    date_formatted = date.strftime(f'%Y%m%d{run_hour}')
+
+    # Pre-build paths and add projection for each model
+    if model == 'GDPS':
+        inter_path = (
+            f'/model_gem_global/15km/grib2/lat_lon/{run_hour}/{forecast_hour}'
+        )
+        file_name = f'/CMC_glb_{{info}}_ISBL_*_latlon.15x.15_{date_formatted}_P{forecast_hour}.grib2'  # noqa
+        first_value = f'/CMC_glb_{{info}}_TGL_{{height}}_latlon.15x.15_{date_formatted}_P{forecast_hour}.grib2'  # noqa
+        out_proj = '+proj=longlat +datum=WGS84 +no_defs'
+    elif model == 'RDPS':
+        inter_path = (
+            f'/model_gem_regional/10km/grib2/{run_hour}/{forecast_hour}'
+        )
+        file_name = f'/CMC_reg_{{info}}_ISBL_*_ps10km_{date_formatted}_P{forecast_hour}.grib2'  # noqa
+        first_value = f'/CMC_reg_{{info}}_TGL_{{height}}_ps10km_{date_formatted}_P{forecast_hour}.grib2'  # noqa
+        out_proj = '+proj=stere +lat_0=90 +lat_ts=60 +lon_0=249 +x_0=0 +y_0=0 +R=6371229 +units=m +no_def'  # noqa
+    elif model == 'HRDPS':
+        inter_path = (
+            f'/model_hrdps/continental/grib2/{run_hour}/{forecast_hour}'
+        )
+        file_name = f'/CMC_hrdps_continental_{{info}}_ISBL_*_ps2.5km_{date_formatted}_P{forecast_hour}-00.grib2'  # noqa
+        first_value = f'/CMC_hrdps_continental_{{info}}_TGL_{{height}}_ps2.5km_{date_formatted}_P{forecast_hour}-00.grib2'  # noqa
+        out_proj = '+proj=stere +lat_0=90 +lat_ts=60 +lon_0=252 +x_0=0 +y_0=0 +R=6371229 +units=m +no_def'  # noqa
+    else:
+        msg = 'Not a valid model: {}'.format(model)
+        LOGGER.error(msg)
+        raise ValueError(msg)
+
+    output = {
+        'type': 'Feature',
+        'geometry': {'type': 'Point', 'coordinates': [lon, lat]},
+        'properties': {},
+    }
+
+    ABBR = {}
+    if temperature_data or wind_data:
+        output['properties']['pressure_unit'] = 'mbar'
+        if temperature_data:
+            ABBR.update({
+                'TMP': 'air_temperature',
+                'DEPR': 'dew_point_depression',
+                'DPT': 'dew_point_temperature',
```

Comment:
general rule (replace globally in PR): remove trailing commas from end of dict properties.